include/nix/NDSize.hpp

```diff
@@ -104,6 +104,25 @@ class NDSizeBase {
         #endif
     }
 
+    template<typename U>
+    NDSizeBase(std::vector<U> args)
```

Comment:
We shouldn't make a copy of the vector argument, i.e. pass the args as const-ref (`const std::vector<U> &args`)